server/server.go

```diff
@@ -55,3 +77,76 @@ func (serv shipperServer) StreamAcknowledgements(streamReq *pb.StreamAcksRequest
 	}
 	return nil
 }
+
+// Run is a blocking call that starts the gRPC server
+func (serv *shipperServer) Run(cfg config.ShipperConfig, agent client.Client) error {
+	if serv.serverIsStarted {
+		return fmt.Errorf("server is already started")
+	}
+
+	// When there is queue-specific configuration in ShipperConfig, it should
+	// be passed in here.
+	queue, err := queue.New()
+	if err != nil {
+		return fmt.Errorf("couldn't create queue: %w", err)
+	}
+	serv.queue = queue
+
+	serv.out = output.NewConsole(serv.queue)
+	serv.out.Start()
+
+	lis, err := net.Listen("tcp", fmt.Sprintf("localhost:%d", cfg.Port))
+	if err != nil {
+		return fmt.Errorf("failed to listen: %w", err)
+	}
+
+	// Beats won't call the "New*" functions of a queue directly, but instead fetch a queueFactory from the global registers.
+	//However, that requires the publisher/pipeline code as well, and I'm not sure we want that.
+
+	// see shipperServer type declaration, this `once` is just to protect expvar for now.
+	serv.monWrap.Do(
+		func() {
+			monHandler, err := loadMonitoring(cfg, serv.queue)
+			if err != nil {
+				serv.log.Errorf("Error starting monitoring: %w", err)
+				return
+			}
+			serv.monHandler = monHandler
+		})
+
+	var opts []grpc.ServerOption
+	if cfg.TLS {
+		creds, err := credentials.NewServerTLSFromFile(cfg.Cert, cfg.Key)
+		if err != nil {
+			return fmt.Errorf("failed to generate credentials %w", err)
+		}
+		opts = []grpc.ServerOption{grpc.Creds(creds)}
+	}
+	serv.grpcServer = grpc.NewServer(opts...)
+
+	pb.RegisterProducerServer(serv.grpcServer, serv)
+
+	_ = agent.Status(proto.StateObserved_HEALTHY, "Started server, grpc listening", nil)
+	serv.log.Infof("gRPC server is listening on port %d", cfg.Port)
+	serv.serverIsStarted = true
+	return serv.grpcServer.Serve(lis)
+}
+
+// blocking command to stop all the shipper components
+func (serv *shipperServer) Stop() {
+	if !serv.serverIsStarted {
+		return
+	}
+	serv.log.Debugf("Stopping shipper server")
+	serv.grpcServer.GracefulStop()
+	serv.queue.Close()
+	// Don't try to gracefully stop monitoring until we deal with expvar
+	//s.monHandler.End()
```

Comment:
is this still pending? It's not obvious to me from the comment what still needs to happen for this line to be activated

Reply:
Yah, I think we might just want to remove the `expvar` `End()` for now, since there isn't really an easy workaround for it.